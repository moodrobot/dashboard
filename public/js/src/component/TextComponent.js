import React, { PureComponent } from "react";
import PropTypes from "prop-types";

class TextComponent extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      style_text: "category form-control",
      style_title: "category",
      error_message: "off",
      focus: false
    };

    this.returnTextArea = this.returnTextArea.bind(this);
    this.returnTextInput = this.returnTextInput.bind(this);
    this.turnOffFocus = this.turnOffFocus.bind(this);
    this.turnOnFocus = this.turnOnFocus.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  static getDerivedStateFromProps(nextProps) {
    if (!nextProps.hasError) {
      return {
        style_text: "category form-control",
        style_title: "category",
        error_message: "off"
      };
    } else {
      return null;
    }
  }

  //SPECIFICALLY FOR NESTED OBJECTS OF TYPE TEXT AREA
  handleChange(e) {
    this.props.handler(e, this.props.nest);
  }

  returnTextInput() {
    return (
      <input
        //className={"category form-control" + ((this.props.hasError && this.state.style_text) ? " error " : " ") + this.props.size}
        className={this.state.style_text + " " + this.props.size}
        type="text"
        placeholder={this.props.required}
        id={this.props.id}
        value={this.props.value}
        disabled={this.props.disabled}
        required={this.props.required}
        onChange={this.handleChange}
        onFocus={this.turnOnFocus}
        onBlur={this.turnOffFocus}
      />
    );
  }

  returnTextArea() {
    return (
      <textarea
        rows={this.props.row}
        className={this.state.style_text}
        type="text"
        placeholder={this.props.required}
        required={this.props.required}
        id={this.props.id}
        disabled={this.props.disabled}
        value={this.props.value}
        onChange={this.handleChange}
        onFocus={this.turnOnFocus}
        onBlur={this.turnOffFocus}
      />
    );
  }

  turnOnFocus() {
    this.setState({
      style_text: "category form-control",
      style_title: "category",
      error_message: "off",
      focus: true
    });
    this.forceUpdate();
  }

  turnOffFocus() {
    if (this.props.hasError) {
      this.setState({
        style_text: "category form-control error",
        style_title: "category error",
        error_message: "on",
        focus: false
      });
    } else {
      this.setState({ focus: false });
    }
    this.forceUpdate();
  }

  render() {
    //console.log("RENDERED TEXTCOMPONENT");
    return (
      <div className="text-component no-gutters">
        <div className={"col-md-3 col-lg-2 d-inline-block " + (this.props.align_top ? "align-top" : "align-middle")}>
          <label htmlFor={this.props.id} className="label-title">
            <span className={this.state.style_title}>{this.props.title}</span>
          </label>
        </div>

        <div className="col-md-9 col-lg-10 align-middle d-inline-block">
          {this.props.type === "area" ? this.returnTextArea() : this.returnTextInput()}

          <div className={"error-message " + this.state.error_message}>{this.props.error_message}</div>
        </div>
      </div>
    );
  }
}

TextComponent.propTypes = {
  value: PropTypes.string,
  type: PropTypes.string,
  required: PropTypes.string,
  disabled: PropTypes.bool,
  hasError: PropTypes.bool.isRequired,
  handler: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  error_message: PropTypes.string,
  size: PropTypes.string,
  row: PropTypes.number,
  nest: PropTypes.string,
  align_top: PropTypes.string,
  long: PropTypes.bool
};

export default TextComponent;
