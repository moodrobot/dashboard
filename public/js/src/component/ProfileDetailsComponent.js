import React, { Component } from "react";
import TextComponent from "../component/TextComponent";
import DropDownComponent from "../component/DropDownComponent";
import SubHead from "../component/SubHead";
import PropTypes from "prop-types";

class ProfileDetailsComponent extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    //console.log("RENDERING PROFILE DETAILS");

    const TT = this.props.translateTitle;
    const TE = this.props.translateError;
    const CE = this.props.componentHasError;

    return (
      <div>
        <SubHead title={TT["title"]} />
        <TextComponent
          title={TT["first-name"]}
          id="first_name"
          hasError={CE.first_name.flag}
          required="Required"
          error_message={TE[CE.first_name.message]}
          size="small"
          value={this.props.data.first_name}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["last-name"]}
          id="last_name"
          hasError={CE.last_name.flag}
          required="Required"
          size="small"
          error_message={TE[CE.last_name.message]}
          value={this.props.data.last_name}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["display-name"]}
          id="display_name"
          hasError={CE.display_name.flag}
          required="Required"
          size="small"
          error_message={TE[CE.display_name.message]}
          value={this.props.data.display_name}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["email"]}
          id="email"
          size="small"
          hasError={CE.email.flag}
          required="Required"
          error_message={TE[CE.email.message]}
          value={this.props.data.email}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["company"]}
          id="company"
          size="small"
          hasError={CE.company.flag}
          error_message={TE[CE.company.message]}
          value={this.props.data.company}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["phone-number"]}
          id="phone"
          size="small"
          hasError={CE.phone.flag}
          error_message={TE[CE.phone.message]}
          value={this.props.data.phone}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <DropDownComponent
          title={TT["country"]}
          data={this.props.countries}
          id="country"
          defaultValue={this.props.data.country}
          onChangeHandler={this.props.handleProfileChange}
          value={this.props.data.country}
          disabled={this.props.freezeAll}
        />
        <DropDownComponent
          title={TT["language-preference"]}
          //data={t("profile-details-language-preference-options", {
          //  returnObjects: true
          //})}
          data={TT["language-preference-options"]}
          id="language"
          value={this.props.currentLanguage}
          defaultValue={this.props.data.language}
          onChangeHandler={this.props.handleLanguageChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["about"]}
          id="bio"
          hasError={CE.bio.flag}
          error_message={TE[CE.bio.message]}
          type="area"
          row={6}
          align_top="true"
          value={this.props.data.bio}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["signature"]}
          id="signature"
          hasError={CE.signature.flag}
          error_message={TE[CE.signature.message]}
          value={this.props.data.signature}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <SubHead title={TT["connect-title"]} />
        <TextComponent
          title={TT["connect-twitter"]}
          id="twitter"
          hasError={CE.twitter.flag}
          error_message={TE[CE.twitter.message]}
          nest="other_services"
          value={(this.props.data.other_services["twitter"]) ? (this.props.data.other_services["twitter"]) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["connect-github"]}
          id="github"
          hasError={CE.github.flag}
          error_message={TE[CE.github.message]}
          nest="other_services"
          value={(this.props.data.other_services.github) ? (this.props.data.other_services.github) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["connect-developer-mesh"]}
          id="developer-mesh"
          hasError={CE["developer-mesh"].flag}
          error_message={TE[CE["developer-mesh"].message]}
          nest="other_services"
          value={(this.props.data.other_services["developer-mesh"]) ? (this.props.data.other_services["developer-mesh"]) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
        <TextComponent
          title={TT["connect-website"]}
          id="website"
          hasError={CE.website.flag}
          error_message={TE[CE.website.message]}
          nest="other_services"
          value={(this.props.data.other_services.website) ? (this.props.data.other_services.website) : ""}
          handler={this.props.handleProfileChange}
          disabled={this.props.freezeAll}
        />
      </div>
    );
  }
}

ProfileDetailsComponent.propTypes = {
  data: PropTypes.object,
  freezeAll: PropTypes.bool,
  translateTitle: PropTypes.object.isRequired,
  translateError: PropTypes.object.isRequired,
  countries: PropTypes.array,
  componentHasError: PropTypes.object,
  handleProfileChange: PropTypes.func,
  handleLanguageChange: PropTypes.func,
  currentLanguage: PropTypes.string,
  init: PropTypes.bool
};

export default ProfileDetailsComponent;
