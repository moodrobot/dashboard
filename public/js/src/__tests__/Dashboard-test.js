import React from "react";
import { shallow } from "enzyme";
import sinon from "sinon";
import ProfileDetailsComponent from "../component/ProfileDetailsComponent";
import TextComponent from "../component/TextComponent";
import translateData from "../../../../locales/en-US.json";
import profileData from "./__mockData__/profileData.json";
import i18n from "../i18n-for-testing"; // initialized i18next instance for testing

import Dashboard from "../Dashboard";
jest.dontMock("sharp");



//STUBS ALLOWS US TO CAPTURE ALL THE XHTTP CALLS
sinon.stub(Dashboard.prototype, "getProfile");
sinon.stub(Dashboard.prototype, "getProducts");
sinon.stub(Dashboard.prototype, "getCountries");
sinon.stub(Dashboard.prototype, "getContributions");
sinon.stub(Dashboard.prototype, "getPage");

const hpc = sinon.spy(Dashboard.prototype, "handleProfileChange");
const name = "Jim";

// beforeAll(() => {
//   console.log("Dashboard", Dashboard);
// });

describe("<Dashboard />", () => {
  const wrapper = shallow(
    <Dashboard
      i18n={i18n}
    />
  );

  it("expect set state to match snapshot", () => {

    //GIVE THIS OUR STATIC DATA
    wrapper.setState({ profile: profileData[0]});
    wrapper.update();

    /**EXPECT NO CHANGE IN THE SNAPSHOT */
    expect(wrapper).toMatchSnapshot();

  });
  xit("expect handleProfileChange to be called when input is changed", () => {

    //PROFILE DETAILS COMPONENT WRAPPER
    const details = wrapper.find(ProfileDetailsComponent).dive();

    //THE FIRST TEXTCOMPONENT WRAPPER
    const textComp = details.find(TextComponent).first().dive();

    //FIND THE INPUT AND CHANGE THE VALUE
    textComp.find("input").simulate("change", {target: {value : name}});

    //CHECKS TO MAKE SURE HANDLE PROFILE IS CALLED WHEN INPUT TEXT CHANGES
    expect(hpc.called).toBe(true);
    expect(hpc.args[0][0].target.value).toBe(name);

  });
});
