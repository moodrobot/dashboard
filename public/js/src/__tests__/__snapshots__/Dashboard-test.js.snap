// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> expect set state to match snapshot 1`] = `
<div
  className="Dashboard"
>
  <Header
    currentLanguage="en-US"
    showProducts={false}
    translate={
      Object {
        "developer-zone": "Developer Zone",
        "page-title": "Dashboard | Intel® Developer Zone",
        "sign-out": "Sign Out",
        "support": "Support",
        "your-contributions": "Your Contributions",
        "your-products": "Your Products",
        "your-profile": "Your Profile",
      }
    }
  />
  <div
    id="profile"
  >
    <div
      className="section-header"
    >
      <div
        className="width-1440 mx-auto row align-items-center no-gutters padding-left-20"
      >
        <div
          className="col-auto"
        >
          <span
            className="header"
          >
            Edit Your Profile
          </span>
        </div>
        <div
          className="col text-right text-sm-nowrap"
        >
          <button
            className="btn btn-secondary active"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <div
              className="btn-loader"
            >
              <img
                className="btn-loader-sm off"
                src="images/loader.gif"
              />
            </div>
            Cancel
          </button>
          <button
            className="btn btn-primary active"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <div
              className="btn-loader"
            >
              <img
                className="btn-loader-sm off"
                src="images/loader.gif"
              />
            </div>
            Save
          </button>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <h2
        className="title"
      >
        Member Since:
         
        9/22/2014
      </h2>
      View your 
      <a
        href="/en-us/user/"
        key="1"
      >
        Public Profile
      </a>
      <SubHead
        title="Profile Image"
      />
      <ProfileImage
        default_images={
          Object {
            "chosen_image": "",
            "images": Array [
              "https://s3.amazonaws.com/dpi-prod/default-images/avatar-Autonomous-2.jpg",
              "https://s3.amazonaws.com/dpi-prod/default-images/avatar-GD.jpg",
              "https://s3.amazonaws.com/dpi-prod/default-images/avatar-IoT-2.jpg",
              "https://s3.amazonaws.com/dpi-prod/default-images/bot01-square.jpg",
              "https://s3.amazonaws.com/dpi-prod/default-images/paper-bird.jpg",
              "https://s3.amazonaws.com/dpi-prod/default-images/wire-head-square.jpg",
            ],
          }
        }
        freezeInput={[Function]}
        profile_image="https://s3.amazonaws.com/idz-profile-images-test/11531542/sirloin-1200-1200.jpg"
        recommendedSize="Recommended: 300x500 pixels"
        remove="Remove"
        setProfileImage={[Function]}
        upload="Upload"
      />
      <ProfileDetailsComponent
        componentHasError={
          Object {
            "bio": Object {
              "flag": false,
              "message": "",
            },
            "company": Object {
              "flag": false,
              "message": "",
            },
            "developer-mesh": Object {
              "flag": false,
              "message": "",
            },
            "display_name": Object {
              "flag": false,
              "message": "",
            },
            "email": Object {
              "flag": false,
              "message": "",
            },
            "first_name": Object {
              "flag": false,
              "message": "",
            },
            "github": Object {
              "flag": false,
              "message": "",
            },
            "last_name": Object {
              "flag": false,
              "message": "",
            },
            "phone": Object {
              "flag": false,
              "message": "",
            },
            "signature": Object {
              "flag": false,
              "message": "",
            },
            "twitter": Object {
              "flag": false,
              "message": "",
            },
            "website": Object {
              "flag": false,
              "message": "",
            },
          }
        }
        countries={Array []}
        currentLanguage="en-US"
        data={
          Object {
            "belt": "blue",
            "bio": "All about Brent - he likes Astoria.

<b>yay?</b>",
            "company": "Intel",
            "country": "US",
            "create_date_timestamp": 1411419675799,
            "display_name": "Brent Rieck (Intel)",
            "email": "brent.rieck@intel.com",
            "enterprise_id": "11531542",
            "first_name": "Brent",
            "language": "en-US",
            "last_name": "Rieck",
            "other_ids": Object {
              "idz": "1083371",
            },
            "other_services": Object {
              "developer-mesh": "bsr",
              "github": "bsr",
              "twitter": "bsr",
              "website": "http://bsr.fun",
            },
            "phone": "503.503.5400",
            "points": "140",
            "profile_image": "https://s3.amazonaws.com/idz-profile-images-test/11531542/sirloin-1200-1200.jpg",
            "signature": "Signature",
            "total_points": "1365",
          }
        }
        freezeAll={false}
        handleLanguageChange={[Function]}
        handleProfileChange={[Function]}
        translateError={
          Object {
            "devmesh": "Must be a valid Developer Mesh username.",
            "email": "Must be a valid email address.",
            "empty": "This field can't be empty.",
            "github": "Must be a valid GitHub username.",
            "noHTML": "HTML tags are not allowed.",
            "required": "This field is required.",
            "someHTML": "The only tags allowed are: <a> <b> <blockquote> <br> <em> <i> <img> <li> <ol> <p> <pre> <s> <strong> <sub> <sup> <ul></pre>",
            "twitter": "Must be a valid Twitter username.",
            "url": "Must be a valid URL.",
          }
        }
        translateTitle={
          Object {
            "about": "About (Public)",
            "company": "Company",
            "connect-developer-mesh": "Developer Mesh",
            "connect-github": "GitHub",
            "connect-linkedin": "LinkedIn",
            "connect-title": "Connect (Public)",
            "connect-twitter": "Twitter",
            "connect-website": "Website",
            "country": "Country/Region",
            "display-name": "Display Name (Public)",
            "email": "Email Address",
            "first-name": "First Name",
            "language-preference": "Language Preference",
            "language-preference-options": Array [
              Object {
                "name": "English",
                "value": "en-US",
              },
              Object {
                "name": "简体中文",
                "value": "zh-CN",
              },
              Object {
                "name": "Español",
                "value": "es-ES",
              },
              Object {
                "name": "Português",
                "value": "pt-BR",
              },
              Object {
                "name": "Русский",
                "value": "ru-RU",
              },
            ],
            "last-name": "Last Name",
            "phone-number": "Phone Number",
            "signature": "Signature (Public)",
            "title": "Profile Details",
          }
        }
      />
      <SubHead
        title="Your Contributions to the Intel Developer Zone"
      />
      <ContributeButtons
        addArticle="Add an Article"
        addBlog="Add a Blog Post"
        addForum="Add a Forum Topic"
        currentLanguage="en-US"
        idzUserId="1083371"
        viewContributions="View All Contributions"
      />
      <hr />
      <BeltStatusComponent
        belt="blue"
        currentLanguage="en-US"
        points="140"
        total_points="1365"
        translate={
          Object {
            "information": "<a href=\\"https://software.intel.com/en-us/blackbelt\\">Learn about belt status &gt;</a>",
            "next-belt-level--title": "Next Belt Level (Green):",
            "next-belt-level-brown-title": "Next Belt Level (Red):",
            "next-belt-level-green-title": "Next Belt Level (Brown):",
            "next-belt-level-points": "Points",
            "next-belt-level-red-title": "",
            "points-activity": "Points Activity",
            "status-": "You don't have a belt yet",
            "status-black": "You are a black belt",
            "status-blue": "You are a blue belt",
            "status-brown": "You are a brown belt",
            "status-green": "You are a green belt",
            "status-red": "You are a red belt",
            "status-title": "Current Status Points:",
            "total-title": "Total Lifetime Points:",
          }
        }
      />
    </div>
    <Footer
      currentLanguage="en-US"
      translate={
        Object {
          "01-org": "01.org",
          "clear-linux-project": "Clear Linux* Project",
          "cookies": "Cookies",
          "download-center": "Download Center",
          "firmware": "Firmware",
          "forums": "Forums",
          "header-hardware-developers": "Hardware Developers",
          "header-manage-your-tools": "Manage Your Tools",
          "header-open-source": "Open Source",
          "header-stay-up-to-date": "Stay Up-to-Date",
          "newsletter-archives": "Newsletter Archives",
          "online-serivce-center": "Online Service Center",
          "privacy": "Privacy",
          "recent-updates": "Recent Updates",
          "registration-center": "Registration Center",
          "resource-and-design-center": "Resource and Design Center",
          "shop-intel": "Shop Intel",
          "subscribe-to-our-youtube-channel": "Subscribe to Our YouTube* Channel",
          "terms-of-use": "Terms of Use",
          "trademarks": "*Trademarks",
          "zephyr-project": "Zephyr Project",
        }
      }
    />
  </div>
  <div
    id="contributions"
  >
    <div
      className="section-header"
    >
      <div
        className="width-1440 mx-auto col-auto padding-left-20"
      >
        <span
          className="header"
        >
          Contributions to the Intel Developer Zone
        </span>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="col-md-12"
      >
        <ContributeButtons
          addArticle="Add an Article"
          addBlog="Add a Blog Post"
          addForum="Add a Forum Topic"
          currentLanguage="en-US"
          idzUserId="1083371"
          viewContributions="View All Contributions"
        />
      </div>
      <ContributionsComponent
        currentLanguage="en-US"
        data={
          Array [
            Object {
              "init": true,
            },
          ]
        }
        init={false}
        translate={
          Object {
            "add-article-button": "Add an Article",
            "add-blog-post-button": "Add a Blog Post",
            "add-forum-topic-button": "Add a Forum Topic",
            "event-names": Object {
              "admin": "Admin",
              "app": "App",
              "article": "Article",
              "blog": "Blog",
              "comment": "Comment",
              "courseware": "Courseware",
              "flag": "Flag",
              "forum": "Forum",
              "project": "Project",
            },
            "event-unpublish": "(unpublished)",
            "filter-by": "Filter By...",
            "header": "Contributions to the Intel Developer Zone",
            "manage-your-content-button": "Manage Your Content",
            "points-earned": "Points Earned",
            "posted": "Posted",
            "title": "Title",
            "type": "Type",
          }
        }
        view_more="View More"
      />
    </div>
    <Footer
      currentLanguage="en-US"
      translate={
        Object {
          "01-org": "01.org",
          "clear-linux-project": "Clear Linux* Project",
          "cookies": "Cookies",
          "download-center": "Download Center",
          "firmware": "Firmware",
          "forums": "Forums",
          "header-hardware-developers": "Hardware Developers",
          "header-manage-your-tools": "Manage Your Tools",
          "header-open-source": "Open Source",
          "header-stay-up-to-date": "Stay Up-to-Date",
          "newsletter-archives": "Newsletter Archives",
          "online-serivce-center": "Online Service Center",
          "privacy": "Privacy",
          "recent-updates": "Recent Updates",
          "registration-center": "Registration Center",
          "resource-and-design-center": "Resource and Design Center",
          "shop-intel": "Shop Intel",
          "subscribe-to-our-youtube-channel": "Subscribe to Our YouTube* Channel",
          "terms-of-use": "Terms of Use",
          "trademarks": "*Trademarks",
          "zephyr-project": "Zephyr Project",
        }
      }
    />
  </div>
  <div
    id="products"
  >
    <div
      className="section-header"
    >
      <div
        className="width-1440 mx-auto row align-items-center padding-left-20"
      >
        <div
          className="col-auto"
        >
          <span
            className="header"
          >
            Manage Your Products
          </span>
        </div>
      </div>
    </div>
    <div
      className="container"
    >
      <ProductsComponent
        currentLanguage="en-US"
        data={
          Array [
            Object {
              "init": true,
            },
          ]
        }
        init={false}
        translate={
          Object {
            "download-button": "Download",
            "expiry-date": "Expiration Date",
            "forum": "Forum",
            "header": "Manage Your Products",
            "license-type": "License Type",
            "manage-license": "Manage License",
            "online-services-button": "Online Services",
            "purchase-date": "Purchase Date",
            "serial": "Serial Number",
          }
        }
        view_more="View More"
      />
    </div>
    <Footer
      currentLanguage="en-US"
      translate={
        Object {
          "01-org": "01.org",
          "clear-linux-project": "Clear Linux* Project",
          "cookies": "Cookies",
          "download-center": "Download Center",
          "firmware": "Firmware",
          "forums": "Forums",
          "header-hardware-developers": "Hardware Developers",
          "header-manage-your-tools": "Manage Your Tools",
          "header-open-source": "Open Source",
          "header-stay-up-to-date": "Stay Up-to-Date",
          "newsletter-archives": "Newsletter Archives",
          "online-serivce-center": "Online Service Center",
          "privacy": "Privacy",
          "recent-updates": "Recent Updates",
          "registration-center": "Registration Center",
          "resource-and-design-center": "Resource and Design Center",
          "shop-intel": "Shop Intel",
          "subscribe-to-our-youtube-channel": "Subscribe to Our YouTube* Channel",
          "terms-of-use": "Terms of Use",
          "trademarks": "*Trademarks",
          "zephyr-project": "Zephyr Project",
        }
      }
    />
  </div>
</div>
`;
